spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/Spark3Util.java

```diff
@@ -745,9 +750,11 @@ public static TableIdentifier identifierToTableIdentifier(Identifier identifier)
    * @param spark a Spark session
    * @param rootPath a table identifier
    * @param format format of the file
+   * @param partitionFilter partitionFilter of the file
    * @return all table's partitions
    */
-  public static List<SparkPartition> getPartitions(SparkSession spark, Path rootPath, String format) {
+  public static List<SparkPartition> getPartitions(SparkSession spark, String tableName, Path rootPath, String format,
```

Comment:
Because this was a public method, I think we need to maintain the API and add the previous version back with a default `partitionFilter`.

Reply:
Good catch. We should assume people might be using anything in Spark3Util in some additional way and not break the API if it can be avoided.

Reply:
Actually, I think I was wrong here, since this is in Spark3Util. I was thinking it was in a different place that is more public. This is okay.